hopper/src/main/java/org/atomhopper/AtomHopperVersionServlet.java

```diff
@@ -19,18 +19,35 @@ public class AtomHopperVersionServlet extends HttpServlet {
     private static final Logger LOG = LoggerFactory.getLogger(AtomHopperVersionServlet.class);
     private static final String POM_PROPERTIES_LOCATION = "META-INF/maven/org.atomhopper/atomhopper/pom.properties";
 
+//    private Properties loadProperties() {
+//        Properties properties = new Properties();
+//        try {
+//            InputStream inStream = getServletContext().getResourceAsStream("/" + POM_PROPERTIES_LOCATION);
+//            properties.load(inStream);
+//            inStream.close();
+//        } catch (Exception e){
+//            LOG.error("Unable to load pom.properties", e);
+//        }
+//        return properties;
+//    }
+
     private Properties loadProperties() {
         Properties properties = new Properties();
-        try {
-            InputStream inStream = getServletContext().getResourceAsStream(POM_PROPERTIES_LOCATION);
-            properties.load(inStream);
-            inStream.close();
-        } catch (Exception e){
+        try (InputStream inStream = getClass().getClassLoader()
+                .getResourceAsStream("META-INF/maven/org.atomhopper/atomhopper/pom.properties");
+        ) {
+            if (inStream == null) {
+                LOG.error("pom.properties resource not found at " + POM_PROPERTIES_LOCATION);
+            } else {
```

Comment:
Won't this just throw an exception that is caught? If this is an attempt to be more verbose with a specific kind of error, add a `catch` block before the generic exception for the specific error you're expecting.